### PR TITLE
Sets up cloudinary + uploads profile picture

### PR DESCRIPTION
**Pull request by webdever12**
### What this PR does (required):
- Sets up Cloudinary on the server
- Allows the user to get to the **Profile Page**
- The user is then given the option to choose a **Profile Picture**
- The picture is then sent to the back-end through **base64EncodedImage** URL
- The image is then sent to Cloudinary 

### Screenshots / Videos (required):
- 
https://user-images.githubusercontent.com/77899847/129404052-7b8471b8-829b-428c-bd8d-b0dcc3c7aab4.mp4

### Any information needed to test this feature (required):
- **npm install cloudinary**  on the server
- inside the **.env** file add:
CLOUDINARY_NAME= 
CLOUDINARY_API_KEY= 
CLOUDINARY_API_SECRET= 
- To get the keys, setup an account in Cloudinary
- In Cloudinary go to Settings => Upload , Add Upload Preset
image
- Then go to => Media Library and create a folder, name the folder **dev_setups**
image

### Any issues with the current functionality (optional):
- Right now the avatar on the top right corner does not get updated when the profile picture is upload

### CODE REVIEW

File: client/src/components/ProfilePictureUpload/Upload.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        />
+        <label htmlFor="raised-button-file">
+          <PublishOutlinedIcon className={classes.chooseimageButton} />
+        </label>
+        <Grid className={classes.submitButton}>
+          <Button variant="outlined" color="primary" type="submit">
+            Submit
+          </Button>
+        </Grid>
+      </form>
```
Comment (moffatethan) [inline client/src/components/ProfilePictureUpload/Upload.tsx]: Anyway you could use Formik here or perhaps React Dropzone?

File: client/src/components/ProfilePictureUpload/Upload.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    if (!previewSource) return;
+    uploadImage(previewSource);
+  };
+
+  const uploadImage = async (base64EncodedImage: string) => {
+    try {
+      await fetch('/api/upload', {
+        method: 'POST',
+        body: JSON.stringify({ data: base64EncodedImage }),
+        headers: { 'Content-type': 'application/json' },
```
Comment (moffatethan) [inline client/src/components/ProfilePictureUpload/Upload.tsx:38]: This isn't the right content type, we should be doing multipart here?

File: client/src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  useEffect(() => {
+    initSocket();
+  }, [initSocket]);
+
+  if (loggedInUser === undefined) return <CircularProgress />;
+  if (!loggedInUser) {
+    history.push('/login');
+
+    return <CircularProgress />;
+  }
```
Comment (moffatethan) [inline client/src/pages/Profile/Profile.tsx]: If we have `ProtectedRoute` we really don't need to do this check if we let protected route handle that for us.

File: client/src/App.tsx [hunk 4]
```diff
@@ -22,6 +23,7 @@ function App(): JSX.Element {
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
                 <ProtectedRoute exact path="/dashboard" component={Dashboard} />
+                <Route exact path="/profile" component={Profile} />
```
Comment (moffatethan) [inline client/src/App.tsx]: Should be protected route and not a regular route.

**Review by moffatethan — COMMENTED** [review]
Left some comments for you, otherwise, this looks alright. I would suggest using multipart form and react dropzone or Formik at least for some easier form management

File: client/src/App.tsx (hunk 4 above)
Comment (webdever12) [reply client/src/App.tsx]: updated!

File: client/src/components/ProfilePictureUpload/Upload.tsx (hunk 2 above)
Comment (webdever12) [reply client/src/components/ProfilePictureUpload/Upload.tsx:38]: updated, see the changes!

**Review by webdever12 — COMMENTED** [review]
All changes made. /ProfileNavBar is deleted. Using one consistent navbar (the main navbar)

**Review by moffatethan — APPROVED** [review]

**Pull request merged**